compiler/rustc_typeck/src/check/inherited.rs

```diff
@@ -89,7 +93,29 @@ impl<'tcx> Inherited<'_, 'tcx> {
             infcx: tcx
                 .infer_ctxt()
                 .ignoring_regions()
-                .with_fresh_in_progress_typeck_results(hir_owner),
+                .with_fresh_in_progress_typeck_results(hir_owner)
+                .with_normalize_fn_sig_for_diagnostic(Lrc::new(move |infcx, fn_sig| {
+                    if fn_sig.has_escaping_bound_vars() {
+                        return fn_sig;
+                    }
+                    infcx.probe(|_| {
+                        let traits::Normalized { value: normalized_fn_sig, obligations } =
+                            traits::normalize(
+                                &mut traits::SelectionContext::new(infcx),
+                                // FIXME(compiler-errors): This is probably not the right param-env...
+                                infcx.tcx.param_env(def_id),
```

Comment:
it mostly is :grin: until we add required bounds to binders

should be fine :shrug: